### ISSUE TITLE
Latitude / Longitude GA4

### ISSUE THREAD

**Issue by PaulieScanlon** [googleanalytics/ga-dev-tools#962]
**What page is this happening on**
missing `ga:latitude,ga:longitude` in GA4 API

**Describe the bug**
Hi team, in the UA API i used to be able to query for `,ga:latitude,ga:longitude`. Was this removed from GA4... i miss it. It was cool and allowed me to plot positions of my users around a sweet ass 3D globe. I'd like to do the same thing with my new GA4 data.

**Comment by keke78ui9** [comment]
Base on this document from [GA4](https://support.google.com/analytics/answer/12002752?hl=en&utm_id=ad), if enable `Granular location and device data collection` should see the Latitude/Longitude, I had enabled but don't see any Latitude/Longitude data, (checked at the event_parameters at BQ).

**Comment by PaulieScanlon** [comment]
Oooh, that's fantastic! Thanks. I'll give it a try.

**Comment by jmcglone** [comment]
Any updates on this @PaulieScanlon ? I'm trying to update a similar usage map application to work with the GA4 API and just realizing the lat/long values aren't available. Wondering if you came up with any workarounds to get the lat/long data.

Example: https://publishing.umich.edu/our-reach

**Comment by jeffdyke** [comment]
Bump

**Comment by InnovexIT-Syl** [comment]
Any Update on this?

**Comment by PJKW999** [comment]
Any updates? As not available from the Web UI either.

**Comment by donaldinou** [comment]
hi, Anyone got a solution for this?

**Comment by rmzonl** [comment]
Hello. Is there anyone who has found a solution for this issue? I tried to get the user’s coordinates with customDimension but I couldn’t succeed. Can someone please help me?

**Comment by Silaszur** [comment]
bump

**Comment by PaulieScanlon** [comment]
I don't think this is every gonna make it into to GA4. I rolled my own solution in the end: [Roll Your Own Analytics With Astro, Vercel Edge Functions and Neon](https://neon.tech/blog/roll-your-own-analytics-with-astro-vercel-edge-functions-and-neon).

It's how all the analytics on my [/dashboard](https://www.paulie.dev/dashboard/) work.

**Comment by Silaszur** [comment]
> I don't think this is every gonna make it into to GA4. I rolled my own solution in the end: [Roll Your Own Analytics With Astro, Vercel Edge Functions and Neon](https://neon.tech/blog/roll-your-own-analytics-with-astro-vercel-edge-functions-and-neon).
> 
> It's how all the analytics on my [/dashboard](https://www.paulie.dev/dashboard/) work.

Thanks for the reply, and for your blog. It is such a shame that they removed this functionality but keeping the city dimension (Maybe they want us to use their geocoding api). They could have just given the coordinates of the city at this point.
